Add suggestion when error happend by using TFLiteConverter.from_keras_model_file

theangels opened the pull request:
If the model has it own loss function, the processing of the transforming will failed. So I add the hint to let them try to use TFLiteConverter.from_session instead of using TFLiteConverter.from_keras_model_file. This hint was suggested by Googler in my previous PR (rejected), so I want to share it.And this PR will not changes the behavior of the converter for every user.

@wangtz

theangels committed:
Update lite.py

If the model has it own loss function, the processing of the transforming will failed. So I add the hint to let them try to use TFLiteConverter.from_session instead of using TFLiteConverter.from_keras_model_file. This hint was suggested by Googler in my previous PR, so I want to share it.And this PR will not changes the behavior of the converter for every user.

gargn commented on tensorflow/lite/python/lite.py:
```diff
@@ -597,23 +597,28 @@ def from_keras_model_file(cls,
-    keras_model = _keras.models.load_model(model_file)
-    sess = _keras.backend.get_session()
+    
+    try:
```
Please make the following changes:
1. Put the exception only around `keras_model = _keras.models.load_model(model_file)`
2. Make sure no line has over 80 characters
3. Use a more specific exception than "Exception"
4. Add a test case

gargn replied:
After thinking about it more, it is not guaranteed that the reason `load_model` fails is because custom parameters are allowed. If someone passes in an invalid file, then `load_model` fails as well. Please make sure to retain and print out the original message first and then add a suggestion to the user that it might be an issue with custom objects.

theangels replied:
ok, I'll follow your suggestion.

theangels committed:
Update lite.py

1. Put the exception only around keras_model = _keras.models.load_model(model_file)
2. Make sure no line has over 80 characters
3. Use a more specific exception than "Exception"
4. Add a test case

theangels committed:
Update lite.py

graph_def = _freeze_graph(sess, output_tensors) -> graph_def = _freeze_graph(sess, input_tensors, output_tensors)

theangels committed:
Merge branch 'master' into master

gargn commented:
After further thought, we have resolved the primary source of the issue - with adding support for `custom_objects` parameter. I think additionally adding support for `compile` parameter flag would be the preferred approach over an additional try/except.

theangels commented:
OK, I see. The conventor v2 skip the load keras model step, maybe it can avoid the loading function errors.
